source/utilities.cc

```diff
@@ -2200,6 +2212,49 @@ namespace aspect
         return 0.0;
     }
 
+    template <int dim>
+    Tensor<1,dim-1>
+    AsciiDataBoundary<dim>::vector_gradient (const types::boundary_id             boundary_indicator,
+                                             const Point<dim>                    &position,
+                                             const unsigned int                   component) const
+    {
+      if ((dynamic_cast<const GeometryModel::Chunk<dim>*>(&this->get_geometry_model()) != nullptr &&
+           dynamic_cast<const InitialTopographyModel::AsciiData<dim>*>(&this->get_initial_topography_model()) != nullptr &&
+           !(this->time_is_initialized())) ||
+          this->get_time() - first_data_file_model_time >= 0.0 )
+        {
+          Point<dim> internal_position = position;
+
+          if (dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()) != 0
+              || dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model()) != 0)
+            {
+              const std::array<double,dim> spherical_position =
+                ::aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(position);
+
+              for (unsigned int i = 0; i < dim; i++)
+                internal_position[i] = spherical_position[i];
+            }
+
+          const std::array<unsigned int,dim-1> boundary_dimensions =
+            get_boundary_dimensions(boundary_indicator);
+
+          Point<dim-1> data_position;
+          for (unsigned int i = 0; i < dim-1; i++)
+            data_position[i] = internal_position[boundary_dimensions[i]];
+
+          const Tensor<1,dim-1>  gradients = lookups.find(boundary_indicator)->second->get_gradients(data_position,component);
+
+          if (!time_dependent)
+            return gradients;
+
+          const Tensor<1,dim-1> old_gradients = old_lookups.find(boundary_indicator)->second->get_gradients(data_position,component);
+
+          return time_weight * gradients + (1 - time_weight) * old_gradients;
+        }
+      else
+        return Tensor<1,dim-1>();
```

Comment:
This way you silently return 0.0 for other geometries than the Chunk? Please make this an `AssertThrow(false, ...)` so that nobody accidentally calls this function for other geometries.

Reply:
I think return this Tensor makes sense for the other condition (`this->get_time() - first_data_file_model_time >= 0.0`). Then if we're not far enough in time yet, 0 is returned. The same is done for `get_data_component`.